### PR TITLE
Add edges for VAP and VAPB

### PR DESCRIPTION
**Pull request by mprahl**
A ValidationAdmissionPolicyBinding now has an edge to its ValidatingAdmissionPolicy. Additionally, all paramRefs are also added as edges to the ValidationAdmissionPolicyBinding.

Relates:
https://issues.redhat.com/browse/ACM-15249

### CODE REVIEW

**Review by yiraeChristineKim — APPROVED** [review]

File: pkg/transforms/vap.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+	typeMeta := metav1.TypeMeta{
+		Kind:       v.GetKind(),
+		APIVersion: v.GetAPIVersion(),
+	}
+
+	apiGroupVersion(typeMeta, &node)
+
+	node.Properties["paramKind_kind"], _, _ = unstructured.NestedString(v.Object, "spec", "paramKind", "kind")
+	node.Properties["paramKind_apiVersion"], _, _ = unstructured.NestedString(v.Object, "spec", "paramKind", "apiVersion")
```
Comment (jlpadilla) [inline pkg/transforms/vap.go:212]: If there's no special logic, I prefer to use the genericResourceConfig.
https://github.com/stolostron/search-collector/blob/main/pkg/transforms/genericResourceConfig.go#L32

**Comment by mprahl** [comment]
@jlpadilla could you please review?

File: pkg/transforms/vapbinding.go [hunk 2]
```diff
@@ -51,6 +89,105 @@ func (v VapBindingResource) BuildNode() Node {
...
+	if v.paramRef == nil {
+		return edges
+	}
+
+	policy, ok := ns.ByKindNamespaceName["ValidatingAdmissionPolicy"]["_NONE"][policyName]
+	if !ok {
+		return edges
+	}
+
+	paramKind, ok := policy.Properties["paramKind_kind"].(string)
```
Comment (jlpadilla) [inline pkg/transforms/vapbinding.go:118]: The [node object has a Metadata field](https://github.com/stolostron/search-collector/blob/main/pkg/transforms/transformer.go#L60-L65) that should be used to store any data needed to build the edges.
The difference is that 
- anything stored in `node.Properties` is sent to the hub and becomes a searchable field.
- data stored in `node.Metadata` is kept within the collector only, so this is used for things like building edges.

Comment (mprahl) [reply pkg/transforms/vapbinding.go]: Good point. I updated the generic transformer configuration to allow metadata only fields. Let me know what you think!

**Review by jlpadilla — APPROVED** [review]
/lgtm

**Comment by mprahl** [comment]
@jlpadilla the PR is rebased. Could you please take another look?

File: pkg/transforms/vapbinding.go [hunk 3]
```diff
@@ -51,6 +89,105 @@ func (v VapBindingResource) BuildNode() Node {
...
+	}
+
+	paramGV, err := schema.ParseGroupVersion(paramApiVersion)
+	if err != nil {
+		return edges
+	}
+
+	namespaceToName := ns.ByKindNamespaceName[paramKind]
+
+	if v.paramRef.Namespace != "" {
```
Comment (yiraeChristineKim) [inline pkg/transforms/vapbinding.go:135]: If paramRef.namesapce is empty or '' (clusterscope), 
```
ns.ByKindNamespaceName[paramKind]["_NONE"] need?
```

Comment (mprahl) [reply pkg/transforms/vapbinding.go:135]: It's just an optimization when the namespace is specified. If it's not specified then all namespaces, including cluster scoped objects in `_NONE` are searched.

**Review by jlpadilla — APPROVED** [review]
/lgtm

**Pull request merged**